src/gatling/simulations/uk/gov/hmcts/ccd/corecasedata/scenarios/manageCaseAssignments/casesharing.scala

```diff
@@ -0,0 +1,259 @@
+import uk.gov.hmcts.ccd.corecasedata.scenarios.utils.Environment
+
+object casesharing {
+
+  val config: Config = ConfigFactory.load()
+  val xuiClientId = "xuiwebapp"
+  val xuiScope = "openid profile roles manage-user create-user manage-roles search-user"
+  val ccdClientId = "ccd_gateway"
+  val ccdScope = "openid profile authorities acr roles openid profile roles"
+  val feedCSUserDataLarge = csv("CaseSharingUsers_Large.csv").circular
+  val feedCSUserDataSmall = csv("CaseSharingUsers_Small.csv").circular
+  val mcaUrl = "http://aac-manage-case-assignment-perftest.service.core-compute-perftest.internal"
+
+  val headers_0 = Map( //Authorization token needs to be generated with idam login
+    "Authorization" -> "AdminApiAuthToken ",
+    "Content-Type" -> "application/json")
+
+  val xuiwebappClientSecret = config.getString("xuiwebappCS")
+  val ccdGatewayClientSecret = config.getString("auth.ccdGatewayCS")
+
+  val CDSGetRequestLarge =
+
+    feed(feedCSUserDataLarge)
+
+      //    .exec {
+      //        session =>
+      //          println(session("userEmail").as[String])
+      //          session
+      //      }
+
+      .exec(http("01_Create_GetS2SToken")
+        .post(Environment.s2sUrl + "/testing-support/lease")
+        .header("Content-Type", "application/json")
+        .body(StringBody("{\"microservice\":\"ccd_data\"}"))
+        .check(bodyString.saveAs("bearerTokenCreate")))
+
+      .exec(http("01_Create_OIDC01_Authenticate")
+        .post(Environment.idamAPI + "/authenticate")
+        .header("Content-Type", "application/x-www-form-urlencoded")
+        .formParam("username", "${email}")
+        .formParam("password", "${password}")
+        .formParam("redirectUri", Environment.xuiMCUrl)
+        .formParam("originIp", "0:0:0:0:0:0:0:1")
+        .check(status is 200)
+        .check(headerRegex("Set-Cookie", "Idam.Session=(.*)").saveAs("authCookie")))
+
+      .exec(http("01_Create_OIDC02_Authorize_CCD")
+        .post(Environment.idamAPI + "/o/authorize?response_type=code&client_id=" + ccdClientId + "&redirect_uri=https://" + Environment.ccdDataStoreUrl + "/oauth2redirect&scope=" + ccdScope).disableFollowRedirect
+        .header("Content-Type", "application/x-www-form-urlencoded")
+        .header("Cookie", "Idam.Session=${authCookie}")
+        .header("Content-Length", "0")
+        .check(status is 302)
+        .check(headerRegex("Location", "code=(.*)&client_id").saveAs("code")))
+
+      .exec(http("01_Create_OIDC03_Token_CCD")
+        .post(Environment.idamAPI + "/o/token?grant_type=authorization_code&code=${code}&client_id=" + ccdClientId + "&redirect_uri=https://" + Environment.ccdDataStoreUrl + "/oauth2redirect&client_secret=" + ccdGatewayClientSecret)
+        .header("Content-Type", "application/x-www-form-urlencoded")
+        .header("Content-Length", "0")
+        .check(status is 200)
+        .check(jsonPath("$.access_token").saveAs("access_tokenCreate")))
+
+  val CDSGetRequestSmall =
```

Comment:
Since Large & Small look identical (other than different feeders), is it possible to put logic around which feeder to use, rather than duplicate all the code?

Reply:
this code is existing in the repo and not mine - the casesharing.scala file has only been moved into a new folder.

Reply:
I have now removed this file from the repo as the code is being rewritten as part of the manage case PI28